### ISSUE TITLE
Confusing loaded/saved/active sessions in the context of TPM2_GetCapability

### ISSUE THREAD

**Issue by jiazhang0** [tpm2-software/tpm2-tss#593]
Here are the related definitions for loaded/saved/active sessions in spec and code.

```
#define TPM_HT_LOADED_SESSION   (0x02 )  /* Loaded Authorization Session  used only in the context of TPM2_GetCapabilityThis type references both loaded HMAC and loaded policy authorization sessions.   */
#define TPM_HT_SAVED_SESSION    (0x03 )  /* Saved Authorization Session  used only in the context of TPM2_GetCapabilityThis type references saved authorization session contexts for which the TPM is maintaining tracking information.  */
#define TPM_HT_HMAC_SESSION     (0x02 )  /* HMAC Authorization Session  assigned by the TPM when the session is created  */
#define TPM_HT_POLICY_SESSION   (0x03 )  /* Policy Authorization Session  assigned by the TPM when the session is created  */



#define HR_HMAC_SESSION (TPM_HT_HMAC_SESSION << HR_SHIFT )       /*   */
#define HR_POLICY_SESSION       (TPM_HT_POLICY_SESSION << HR_SHIFT )     /*   */

#define HMAC_SESSION_FIRST      (HR_HMAC_SESSION + 0 )   /* first HMAC session  */
#define HMAC_SESSION_LAST       (HMAC_SESSION_FIRST + MAX_ACTIVE_SESSIONS - 1 )  /* last HMAC session  */
#define LOADED_SESSION_FIRST    (HMAC_SESSION_FIRST )    /* used in GetCapability  */
#define LOADED_SESSION_LAST     (HMAC_SESSION_LAST )     /* used in GetCapability  */

#define POLICY_SESSION_FIRST    (HR_POLICY_SESSION + 0 )         /* first policy session  */
#define POLICY_SESSION_LAST     (POLICY_SESSION_FIRST + MAX_ACTIVE_SESSIONS - 1 )        /* last policy session   */
#define ACTIVE_SESSION_FIRST    (POLICY_SESSION_FIRST )  /* used in GetCapability  */
#define ACTIVE_SESSION_LAST     (POLICY_SESSION_LAST )   /* used in GetCapability  */

#define TPM_PT_HR_LOADED_MIN    (PT_FIXED + 16 )         /* the minimum number of authorization sessions that can be held in TPM RAM NOTE       This minimum shall be no less than the minimum value required by the platformspecific specification to which the TPM is built.  */
#define TPM_PT_ACTIVE_SESSIONS_MAX      (PT_FIXED + 17 )         /* the number of authorization sessions that may be active at a timeA session is active when it has a context associated with its handle. The context may either be in TPM RAM or be context saved.NOTE        This value shall be no less than the minimum value required by the platformspecific specification to which the TPM is built.  */
```

I’m confused a bit when I write such a handler of processing the returned handles from TPM2_GetCapability:

```
static int dump_tpm_capability (TPMU_CAPABILITIES *capabilities) {

    switch (options.capability) {
    case TPM_CAP_ALGS:
…
    case TPM_CAP_HANDLES:
        switch (options.property & HR_RANGE_MASK) {
        case HR_TRANSIENT:
        case HR_PERSISTENT:
        case HR_PERMANENT:
        case HR_PCR:
        case HR_NV_INDEX:
        case TPM_HT_LOADED_SESSION << HR_SHIFT:
        case TPM_HT_SAVED_SESSION << HR_SHIFT:
            dump_handles (capabilities->handles.handle,
                          capabilities->handles.count);
            break;
```

First, there is no definitions of HR_* for loaded and saved session. I have to use TPM_HT_LOADED_SESSION << HR_SHIFT and TPM_HT_SAVED_SESSION << HR_SHIFT instead. This should be simpler as other options.properties.

Second, when passing the property parameter of TPM2_GetCapability(), looks like I should use ACTIVE_SESSION_FIRST, but why there is no corresponding definitions for TPM_HT_ACTIVE_SESSION and HR_ACTIVE_SESSION?

In other words, there are conflicting concepts here when talking the context of TPM2_GetCapability(). We need a consistent semantics for the property values 0x02 << HR_SHIFT and 0x03 << HR_SHIFT and similar. Among “loaded”, “saved”, “active” sessions, we should choose two key words for a consistent naming scheme for TPM_HT_*, HR_* and *_FIRST|LAST. However, currently they have different semantics according to the comments. 

I’m confused and not sure how to write a correct property parameter for TPM2_GetCapability() and the handler of processing the returned handle values like above.

**Comment by tstruk** [comment]
These are valid questions, but this is all that the spec defines:
https://trustedcomputinggroup.org/wp-content/uploads/TCG-TSS-2.0-Overview-and-Common-Structures-Specification-Version-0.90-Revision-02.pdf
